File: calculadoras/calculadora.java
import java.util.Scanner;

public class Calculadora {
  public static void main(String[] args) {

    // Criando um objeto Scanner para ler a entrada do usuário

    Scanner input = new Scanner(System.in);

    // Exibindo as opções disponíveis

    System.out.println("Selecione a operação:");
    System.out.println("1. Adição");
    System.out.println("2. Subtração");
    System.out.println("3. Multiplicação");
    System.out.println("4. Divisão");

    // Lendo a escolha do usuário

    System.out.print("Digite a opção (1/2/3/4): ");
    int escolha = input.nextInt();

    // Lendo os números de entrada

    System.out.print("Digite o primeiro número: ");
    double num1 = input.nextDouble();
    System.out.print("Digite o segundo número: ");
    double num2 = input.nextDouble();

    // Realizando a operação escolhida

    double resultado;
    switch (escolha) {
      case 1:
        resultado = num1 + num2;
        break;
      case 2:
        resultado = num1 - num2;
        break;
      case 3:
        resultado = num1 * num2;
        break;
      case 4:
        resultado = num1 / num2;
        break;
      default:
        System.out.println("Opção inválida!");
        return;
    }

    // Exibindo o resultado
    
    System.out.println("O resultado é: " + resultado);
  }
}

File: calculadoras/calculadora.php
<!DOCTYPE html>
<html>
<head>
	<title>Calculadora em PHP</title>
</head>
<body>
	<h1>Calculadora em PHP</h1>
	<form method="post">
		<label for="num1">Digite o primeiro número:</label>
		<input type="number" name="num1" id="num1" required>

		<label for="operacao">Escolha a operação:</label>
		<select name="operacao" id="operacao" required>
			<option value="adicao">Adição (+)</option>
			<option value="subtracao">Subtração (-)</option>
			<option value="multiplicacao">Multiplicação (*)</option>
			<option value="divisao">Divisão (/)</option>
		</select>

		<label for="num2">Digite o segundo número:</label>
		<input type="number" name="num2" id="num2" required>

		<input type="submit" value="Calcular">
	</form>

	<?php
	if ($_SERVER["REQUEST_METHOD"] == "POST") {
		// Lendo os valores de entrada
		$num1 = $_POST["num1"];
		$num2 = $_POST["num2"];
		$operacao = $_POST["operacao"];

		// Realizando a operação escolhida
		switch ($operacao) {
			case "adicao":
				$resultado = $num1 + $num2;
				break;
			case "subtracao":
				$resultado = $num1 - $num2;
				break;
			case "multiplicacao":
				$resultado = $num1 * $num2;
				break;
			case "divisao":
				$resultado = $num1 / $num2;
				break;
			default:
				echo "<p>Opção inválida!</p>";
				return;
		}

		// Exibindo o resultado
		echo "<p>O resultado é: " . $resultado . "</p>";
	}
	?>
</body>
</html>
